lms/static/js/learner_dashboard/models/program_subscription_model.js

```diff
@@ -0,0 +1,87 @@
+import Backbone from 'backbone';
+import moment from 'moment';
+
+import DateUtils from 'edx-ui-toolkit/js/utils/date-utils';
+
+/**
+ * Model for Program Subscription Data.
+ */
+class ProgramSubscriptionModel extends Backbone.Model {
+    constructor({ context }, ...args) {
+        const {
+            subscriptionData: [data = {}],
+            programData: { subscription_prices },
+            urls,
+            userPreferences,
+        } = context;
+
+        const subscriptionState = data.subscription_state?.toLowerCase() ?? '';
+        const subscriptionPrice = '$' + parseFloat(
+            subscription_prices?.find(({ currency }) => currency === 'USD')?.price
+        );
```

Comment:
nit: honestly, I found this a little hard to grok and seems to have a lot going on. would love to see it split up but i'm not used to looking at JS as much as python

Reply:
sure